app/actions/handlers.py

```diff
@@ -14,85 +14,78 @@
 
     for vehicle in vehicles:
-        if vehicle.ats_serial_num not in vehicles_to_skip:
-            # Get GmtOffset for this device
-            gmt_offset_value = gmt_offsets.get(vehicle.ats_serial_num)
-            time_delta = datetime.timedelta(hours=gmt_offset_value)
-            timezone_object = datetime.timezone(time_delta)
+        # Get GmtOffset for this device
+        time_delta = datetime.timedelta(hours=offset)
+        timezone_object = datetime.timezone(time_delta)
 
-            date_year_and_julian_with_tz = vehicle.date_year_and_julian.replace(tzinfo=timezone_object)
+        date_year_and_julian_with_tz = vehicle.date_year_and_julian.replace(tzinfo=timezone_object)
 
-            vehicle.date_year_and_julian = date_year_and_julian_with_tz
+        vehicle.date_year_and_julian = date_year_and_julian_with_tz
 
-            # Get current state for the device
-            current_state = await state_manager.get_state(
-                integration_id,
-                action_id,
-                vehicle.ats_serial_num
+        # Get current state for the device
+        current_state = await state_manager.get_state(
+            integration_id,
+            action_id,
+            vehicle.ats_serial_num
+        )
+
+        if current_state:
+            # Compare current state with new data
+            latest_device_timestamp = datetime.datetime.strptime(
+                current_state.get("latest_device_timestamp"),
+                '%Y-%m-%d %H:%M:%S%z'
             )
 
-            if current_state:
-                # Compare current state with new data
-                latest_device_timestamp = datetime.datetime.strptime(
-                    current_state.get("latest_device_timestamp"),
-                    '%Y-%m-%d %H:%M:%S%z'
+            if vehicle.date_year_and_julian <= latest_device_timestamp:
+                # Data is not new, not transform
+                logger.info(
+                    f"Excluding device ID '{vehicle.ats_serial_num}' obs '{vehicle.date_year_and_julian}'"
                 )
-
-                if vehicle.date_year_and_julian <= latest_device_timestamp:
-                    # Data is not new, not transform
-                    logger.info(
-                        f"Excluding device ID '{vehicle.ats_serial_num}' obs '{vehicle.date_year_and_julian}'"
-                    )
-                    continue
-
-            data = {
-                "source": vehicle.ats_serial_num,
-                "source_name": vehicle.ats_serial_num,
-                'type': 'tracking-device',
-                "recorded_at": vehicle.date_year_and_julian,
-                "location": {
-                    "lat": vehicle.latitude,
-                    "lon": vehicle.longitude
-                },
-                "additional": {
-                    key: value for key, value in vehicle.dict().items()
-                    if key not in main_data and value is not None
-                }
+                continue
```

Comment:
I think we can simply remove this.
I believe it's not necessary, and might be a source of problems.
- the datasource advances the cursor upon reading, so we should not be concerned with reading duplicates
- if we were to get data out of order, then this would exclude a point we have not yet processed. 

In fact, I don't know why we store a latest-timestamp in the state for any of these, so we can probably just remove that too.